streaming/base/converters/dataframe_to_mds.py

```diff
@@ -0,0 +1,326 @@
+        compression (str, optional): Optional compression or compression:level. Defaults to
+            ``None``.
+        hashes (List[str], optional): Optional list of hash algorithms to apply to shard files.
+            Defaults to ``None``.
+        size_limit (Union[int, str], optional): Optional shard size limit, after which point to start a new
+            shard. If ``None``, puts everything in one shard. Can specify bytes
+            human-readable format as well, for example ``"100kb"`` for 100 kilobyte
+            (100*1024) and so on. Defaults to ``1 << 26``
+        udf_iterable (Callable or None): A user-defined function that returns an iterable over the dataframe. ppfn_kwargs is the k-v args for the method. Default is None.
+        udf_kwargs (Dict): Additional keyword arguments to pass to the pandas processing
+            function if provided. Default is an empty dictionary.
+
+    Return:
+        mds_path: actual local and remote path were used
+        fail_count: number of records failed to be converted
+    Raises:
+        ValueError: If dataframe is not provided
+
+    Note:
+        - The method creates a SparkSession if not already available.
+        - If 'sample_ratio' is provided, the input data will be randomly sampled.
+        - The 'ppfn_kwargs' dictionaries can be used to pass additional
+          keyword arguments to the udf_iterable.
+    """
+
+    def write_mds(iterator: Iterable):
+
+        context = TaskContext.get()
+        if context is not None:
+            id = context.taskAttemptId()
+        else:
+            raise Exception('TaskContext.get() returns None')
+
+        if isinstance(mds_path, str):  # local
+            output = os.path.join(mds_path, f'{id}')
+            out_file_path = output
+        else:
+            output = (os.path.join(mds_path[0], f'{id}'), os.path.join(mds_path[1], f'{id}'))
+            out_file_path = output[0]
+
+        if mds_kwargs:
+            kwargs = mds_kwargs.copy()
+            kwargs['out'] = output
+        else:
+            kwargs = {}
+
+        if merge_index:
+            kwargs['keep_local'] = True  # need to keep local to do merge
+
+        count = 0
+
+        with MDSWriter(**kwargs) as mds_writer:
+            for pdf in iterator:
+                if udf_iterable is not None:
+                    records = udf_iterable(pdf, **udf_kwargs or {})
+                else:
+                    records = pdf.to_dict('records')
+                assert is_iterable(
+                    records
+                ), f'pandas_processing_fn needs to return an iterable instead of a {type(records)}'
+
+                for sample in records:
+                    try:
+                        mds_writer.write(sample)
+                    except:
+                        logger.warning(f'failed to write sample: {sample}')
+                        count += 1
+                        raise Exception(f'failed to write sample: {sample}')
```

Comment:
Looks we are throwing the exception again here which breaks the conversion script, is it intentional ?

Reply:
This was for debugging. Removing "Raise Exception()" so the call will keep running and adding warning printout to spark log.